### DISCUSSION TITLE
How to create custom blocks?

### DISCUSSION THREAD

**Discussion by ericraio** [harttle/liquidjs#400] [Working with Shopify]
In Ruby, I was able to create shopify's [paginate block](https://shopify.dev/api/liquid/objects/paginate) by inheriting from Liquid::Block, however I am trying to figure out how to recreate the same block with LiquidJS, this is this a possibility?

**Comment by harttle** [comment]
FYI: https://github.com/harttle/liquidjs/issues/387#issuecomment-933502062

Maybe someone propose a specific use case and we can see how to implement in JavaScript. Looking forward to more ideas/code from you guys :)
